### ISSUE TITLE
Performance issue: browser get slow when the chat log is long

### ISSUE THREAD

**Issue by JohnXLivingston** [JohnXLivingston/peertube-plugin-livechat#140]
**Describe the bug**
On long and intensive chat, there can be thousands of messages.
The load on the browser becomes too high.

**To Reproduce**
1. Have a chat with thousands of messages

**Expected behavior**
Browser should only keep last X message (X = 100? 1000?)

Reference: related to NlNet-task-10

**Comment by jcbrand** [comment]
Converse has a setting `prune_message_above` which can be used to keep only the last X messages.

https://conversejs.org/docs/html/configuration.html#prune-messages-above

**Comment by JohnXLivingston** [comment]
Thanks @jcbrand . I already saw this option. But until now, I though it would not be necessary. But last week, we had a Peertube Live lasting more than 2 hours, with 450 viewers. And browsers began to slow down after few hundred of messages.

I will use this option. Thanks again :)

**Comment by Ppjet6** [comment]
Some other settings it might be worth looking into regarding fetching messages:
- `mam_request_all_pages` turned off
-  As well as a tweaked `archived_messages_page_size`. Maybe tuned down to about the number of messages being displayed on screen (plus a few to make transitions somewhat smooth still). This second setting probably requires some measurement.
- `muc_history_max_stanzas` to 0 as we're using MAM. As the doc says. This will prevent the MUC from sending us N messages (defaults to 20 in prosody) as a response to our join request. Messages that we will get anyway with MAM.

**Comment by JohnXLivingston** [comment]
Note to myself: I have to disable «MUC history», which was an ancestror of «MAM» (Message Archive Management).

MUC history will only serve you N messages on join and that's it. With MAM the client can request the number of messages it wants and when it wants them, and can go up to the first message stored by the MUC.

**Comment by JohnXLivingston** [comment]
Note: i copied the various comments on this issue to this one: https://github.com/JohnXLivingston/peertube-plugin-livechat/issues/96

This current issue (140) will only concern the change for `prune_messages_above`.

**Comment by JohnXLivingston** [comment]
Setting `prune_messages_above` to 100 is confirmed to fix this issue.
I made some tests with my new stress-test tools. See the results here:

https://github.com/JohnXLivingston/livechat-perf-test/tree/main/tests/10-browser-vs-flooding-bot

(runs `2023-12-18`, `2023-12-19`, `03-prune-message-above-100`, `04-prune-message-above-100-converse-patched`)

TL;DR:

Before:
![image](https://github.com/JohnXLivingston/peertube-plugin-livechat/assets/38844060/2adf62b6-a766-42cd-bfa9-064cf417ec45)

After:
![image](https://github.com/JohnXLivingston/peertube-plugin-livechat/assets/38844060/65dc732b-a736-42a7-92fc-5d323a6e3723)
